Wikipedia/Code/WMFExploreViewController.m

```diff
@@ -682,6 +686,14 @@ - (WMFLayoutEstimate)collectionView:(UICollectionView *)collectionView estimated
         } break;
         case WMFFeedDisplayTypePageWithPreview: {
             WMFArticle *article = [self articleForIndexPath:indexPath];
+            NSString *key = article.key;
+            NSString *cacheKey = [NSString stringWithFormat:@"%@-%lli", key, (long long)columnWidth];
+            NSNumber *cachedValue = [self.cachedHeights objectForKey:cacheKey];
+            if (cachedValue) {
+                estimate.height = [cachedValue doubleValue];
```

Comment:
@montehurd  good catch, I addressed this in the separate PR I opened for this change